netbox_netprod_importer/devices_list.py

```diff
@@ -28,3 +31,57 @@ def parse_devices_yaml_def(devices_yaml, creds=None):
                     "Cannot connect to device %s: %s", hostname, e
                 )
     return devices
+
+
+def parse_filter_yaml_def(filter_yaml, creds=None):
+    netbox_api = NetboxAPI(**get_config()["netbox"])
+    devices = {}
+    with open(filter_yaml) as filter_yaml_str:
+        q_list = []
+        yml = yaml.safe_load(filter_yaml_str)
+        for fil, props in tqdm(yml["filter"].items()):
```

Comment:
I don't understand the point of using tqdm here to iterate into the yaml. It's going to be really quick anyway, no?

Reply:
Сopied your code from the parse_devices_yaml_def function

Reply:
Yes, but in `parse_devices_yaml`, in each iteration of the loop, I initialize the Importer for the device, which try to connect to the device (so it can actually take some time with many devices). That's why I added a progression bar.

Here, it will take a few ms at most, I don't thing you need it.

Reply:
Good, but this part of the code is gone :) because yaml is passed directly to netbox_ap.get